src/rebar3_coveralls.erl

```diff
@@ -0,0 +1,79 @@
+%%%      notice, this list of conditions and the following disclaimer in the
+%%%      documentation and/or other materials provided with the distribution.
+%%%    * Neither the name of the <organization> nor the
+%%%      names of its contributors may be used to endorse or promote products
+%%%      derived from this software without specific prior written permission.
+%%%
+%%% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+%%% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+%%% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+%%% ARE DISCLAIMED. IN NO EVENT SHALL MARKUS EKHOLM BE LIABLE FOR ANY
+%%% DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+%%% (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+%%% LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+%%% ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+%%% (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+%%% THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+%%%
+%%% @end
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-module(rebar3_coveralls).
+-behaviour(provider).
+
+-export([init/1, do/1, format_error/1]).
+
+-define(PROVIDER, send).
+-define(DEPS, [{default, app_discovery}]).
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+-spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
+init(State) ->
+  Provider = providers:create([{name,       ?PROVIDER},
+                               {module,     ?MODULE},
+                               {namespace,  coveralls},
+                               {bare,       true},
+                               {deps,       ?DEPS},
+                               {example,    "rebar3 coveralls send"},
+                               {short_desc, "Send coverdata to coveralls."},
+                               {desc,       "Send coveralls to coveralls."},
+                               {opts,       []}]),
+  {ok, rebar_state:add_provider(State, Provider)}.
+
+-spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
+do(State) ->
+  rebar_api:info("Running coveralls...", []),
+  ConvertAndSend = fun coveralls:convert_and_send_file/3,
+  Get            = fun(Key, Def) -> rebar_state:get(State, Key, Def) end,
+  GetLocal       = fun(Key, Def) -> rebar_state:get(State, Key, Def) end,
+  MaybeSkip      = fun() -> ok end,
+  ok = cover_paths(State),
+  rebar_coveralls:do_coveralls(ConvertAndSend, Get, GetLocal, MaybeSkip, 'send-coveralls'),
+  {ok, State}.
+
+-spec format_error(any()) -> iolist().
+format_error(Reason) ->
+    io_lib:format("~p", [Reason]).
+
+cover_paths(State) ->
+    lists:foreach(fun(App) ->
```

Comment:
Indent at 2, not 4.

Add the following footer to the file:
%%% Local Variables:
 %%% allout-layout: t
 %%% erlang-indent-level: 2
 %%% End: